packages/ui/src/bounty/modals/SubmitWorkModal/SubmitWorkModal.tsx

```diff
@@ -62,7 +63,11 @@ export const SubmitWorkModal = () => {
     }
   })
   const transaction = useMemo(() => {
-    const entry = modalData.bounty.entries?.find((entry) => entry.worker.id === activeMember?.id)
+    const entry = modalData.bounty?.entries?.find((entry) => entry.worker.id === activeMember?.id)
+    if (!entry) {
+      hideModal()
+    }
```

Comment:
Nice catch! But we shouldn't have side effects in `useMemo`, please move the `hideModal()` in a `useEffect` (and I guess `modalData.bounty?.entries?.find((...` in a diferent `useMemo`)